File: HRMS.Core/Models/Employee.cs
using HRMS.Core.Enums.Model;

namespace HRMS.Core.Models;

public class Employee
{
    public int Id { get; set; }

    public string FirstName { get; set; } = string.Empty;

    public string LastName { get; set; } = string.Empty;

    public string MaidenName { get; set; } = string.Empty;

    public string ParentName { get; set; } = string.Empty;

    public Gender Gender { get; set; }

    public string RegistrationNumber { get; set; } = string.Empty;

    public string PersonalIdentificationNumber { get; set; } = string.Empty;

    public string WorkerCode { get; set; } = string.Empty;

    public DateTime BirthDate { get; set; }

    public City BirthPlace { get; set; } = new();

    public string Address { get; set; } = string.Empty;

    public City City { get; set; } = new();

    public Country Citizenship { get; set; } = new();

    public string Image { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public string Phone { get; set; } = string.Empty;

    public string Mobile { get; set; } = string.Empty;

    public string OfficePhone { get; set; } = string.Empty;

    public string Profession { get; set; } = string.Empty;

    public Education Education { get; set; } = new();

    public int PreviousLOSYears { get; set; }

    public int PreviousLOSMonths { get; set; }

    public string BankAccount { get; set; } = string.Empty;

    public string Note { get; set; } = string.Empty;

    public DateTime CreateDate { get; set; }
}

File: HRMS.Desktop/lib/models/employee.dart
import 'package:json_annotation/json_annotation.dart';
import 'city.dart';
import 'country.dart';
import 'education.dart';

part 'employee.g.dart';

@JsonSerializable()
class Employee {
  int id;
  String firstName;
  String lastName;
  String maidenName;
  String parentName;
  String registrationNumber;
  String personalIdentificationNumber;
  String workerCode;
  DateTime birthDate;
  City? birthPlace;
  String address;
  City? city;
  Country? citizenship;
  String image;
  String email;
  String phone;
  String mobile;
  String officePhone;
  String profession;
  Education? education;
  int previousLOSYears;
  int previousLOSMonths;
  String bankAccount;
  String note;
  DateTime createDate;

  Employee(
    this.id,
    this.firstName,
    this.lastName,
    this.maidenName,
    this.parentName,
    this.registrationNumber,
    this.personalIdentificationNumber,
    this.workerCode,
    this.birthDate,
    this.birthPlace,
    this.address,
    this.city,
    this.citizenship,
    this.image,
    this.email,
    this.phone,
    this.mobile,
    this.officePhone,
    this.profession,
    this.education,
    this.previousLOSYears,
    this.previousLOSMonths,
    this.bankAccount,
    this.note,
    this.createDate,
  );

  factory Employee.fromJson(Map<String, dynamic> json) =>
      _$EmployeeFromJson(json);

  Map<String, dynamic> toJson() => _$EmployeeToJson(this);
}
